hr_holidays_legal_leave/models/hr_holidays_status.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import fields, models, api
+
+
+class HolidaysType(models.Model):
+    _inherit = "hr.holidays.status"
+
+    is_annual = fields.Boolean(
+        'Legal/Annual',
+        compute='_compute_is_annual',
+        inverse='_inverse_is_annual',
+        help='Use this Leave type as Legal/Annual for current company. '
+             'One and only one leave type can have this checkbox. '
+             'You cannot unset it directly. '
+             'Set it on another Leave type instead. '
+    )
+
+    @api.multi
+    def _compute_is_annual(self):
+        company = self.env.user.company_id
+        for r in self:
+            r.is_annual = r.id == company.legal_holidays_status_id.id
```

Comment:
Uhm, in this case is because of there's no record. Anyway, you can use:
```
record = self.filtered(lamba x: x == company.legal_holidays_status_id)
if record:
    record.is_annual = True
```
or try this:
```
record = self.filtered(lamba x: x == company.legal_holidays_status_id).update({'is_annual': True})
```